### PR TITLE
Migrate CBOR classes to System.Formats.Cbor namespace

### PR DESCRIPTION
**Pull request by eiriktsarpalis**
- [x] Moves cbor classes to the System.Formats.Cbor namespace.
- [x] Mark production Cbor classes public.

### CODE REVIEW

**Comment by bartonjs** [comment]
Please don't do this until after API review.

**Comment by eiriktsarpalis** [comment]
I've updated the PR so the new structure is now hosted in System.Security.Cryptography.Encoding.

File: src/libraries/System.Security.Cryptography.Encoding/src/System.Security.Cryptography.Encoding.csproj [hunk 1]
```diff
@@ -20,66 +20,42 @@
     <Compile Include="System\Security\Cryptography\OidGroup.cs" />
     <Compile Include="System\Security\Cryptography\PemFields.cs" />
     <Compile Include="System\Security\Cryptography\PemEncoding.cs" />
-    <Compile Include="$(CommonPath)Internal\Cryptography\Helpers.cs"
-             Link="Internal\Cryptography\Helpers.cs" />
-    <Compile Include="$(CommonPath)System\Security\Cryptography\CryptoPool.cs"
-             Link="System\Security\Cryptography\CryptoPool.cs" />
-    <Compile Include="$(CommonPath)System\HexConverter.cs"
-             Link="Common\System\HexConverter.cs" />
+    <Compile Include="$(CommonPath)Internal\Cryptography\Helpers.cs" Link="Internal\Cryptography\Helpers.cs" />
```
Comment (bartonjs) [inline src/libraries/System.Security.Cryptography.Encoding/src/System.Security.Cryptography.Encoding.csproj:23]: Please revert all of the changes in this csproj, including the file moves.  It's not necessary to the namespace rename.

Comment (eiriktsarpalis) [reply src/libraries/System.Security.Cryptography.Encoding/src/System.Security.Cryptography.Encoding.csproj]: Done.

File: src/libraries/System.Security.Cryptography.Encoding/tests/Cbor.Tests/CborReaderTests.cs [hunk 2]
```diff
@@ -59,17 +59,17 @@ public static void ConformanceLevel_DefaultValue_ShouldEqualLax()
         }
 
         [Theory]
-        [InlineData(CborMajorType.UnsignedInteger, CborReaderState.UnsignedInteger)]
```
Comment (eiriktsarpalis) [inline src/libraries/System.Security.Cryptography.Encoding/tests/Cbor.Tests/CborReaderTests.cs:62]: Replacing dependency on internal implementation details.

**Review by bartonjs — APPROVED** [review]

**Pull request merged**